Use PSR7 response methods where possible.

markstory opened the pull request:
In Component callbacks that return a response we can start using the PSR7 methods as the event dispatcher passes the returned response up the stack and eventually emits it.

Refs #9636

markstory committed:
Use PSR7 response methods where possible.

In Component callbacks that return a response we can start using the
PSR7 methods as the event dispatcher passes the returned response up the
stack and eventually emits it.

Refs #9636

dereuromark commented on src/Controller/Component/AuthComponent.php:
```diff
@@ -384,13 +385,10 @@ protected function _unauthenticated(Controller $controller)
-            $response->statusCode(403);
-
-            return $response;
+            return $response->withStatus(403);
```
Do we even need that line if the next one returns the $response then anway like this?

markstory committed:
Update Controller::redirect() as well.

In addition to the Controller method, fixup the Stub\Response to work
better with PSR7 responses. By returning $this, we can start using
immutable methods and not adversely effect tests. Without this if people
used immutable methods their tests would start failing.

markstory committed:
Fix PHPCS errors.

markstory committed:
Fix PHPCS error.
